WordPress/src/main/java/org/wordpress/android/viewmodel/posts/PostListItemUiStateHelper.kt

```diff
@@ -381,7 +381,7 @@ class PostListItemUiStateHelper @Inject constructor(
                 !isLocalDraft &&
                 !isLocallyChanged
         val canShowCopy = postStatus == PUBLISHED || postStatus == DRAFT
-        val canShowCopyUrlButton = postStatus == PUBLISHED && !isLocalDraft
+        val canShowCopyUrlButton = !isLocalDraft && postStatus != TRASHED
```

Comment:
@antonis I checked in the wp-calypso source code and [there](https://github.com/Automattic/wp-calypso/blob/c2a3f57266c1bb6123cd9b474491c31f42fca6a9/client/my-sites/post-type-list/post-actions-ellipsis-menu/index.jsx#L21) we show the copy link button regardless of the post status.
For the mobile apps it wouldn't make sense to show it for local drafts, since this type of post will not have an URL.

I also kept it hidden for trashed posts but we can review this, since on the web version we show it:

<details>
<summary>Web: Copy link for trashed posts</summary>

<img width="860" alt="copy_link_web_trashed" src="https://user-images.githubusercontent.com/4588074/143127450-f6b2cc70-36b8-40c0-96b7-0295f1688870.png">

</details>

Reply:
Thank you for tackling this @ovitrif 🙇 

> For the mobile apps it wouldn't make sense to show it for local drafts

This makes sense 👍 

>I also kept it hidden for trashed posts but we can review this, since on the web version we show it

I think it makes sense to keep it as is since even on the web the copied link leads to a "page not found" error. We can revisit this in the future if there is a need to add the functionality for the trashed posts.